ads/google/csa.js

```diff
@@ -0,0 +1,349 @@
+      const ignore = document.body.offsetHeight;
+      /*eslint-enable no-unused-vars*/
+
+      // Capture new height
+      const container = document.getElementById('csacontainer');
+      let newHeight = container.style.height;
+
+      // In older versions of iOS, this height will be different because the
+      // container height is resized.
+      // In Chrome and iOS 10.0.2 the height is the same because
+      // the container isn't resized.
+      if (oldHeight != newHeight && newHeight != currentAmpHeight) {
+
+        // style.height returns "60px" (for example), so turn this into an int
+        newHeight = parseInt(newHeight, 10);
+
+        // Also update the onclick function to resize to the right height.
+        const overflow = document.getElementById('overflow');
+        if (overflow) {
+          overflow.onclick = function() {
+            window.context.requestResize('auto', newHeight);
+          };
+        }
+
+        // Resize the container to the correct height
+        window.context.requestResize('auto', newHeight);
+      }
+
+    }, orientationChangeTimeout);
+
+  }, false);
+
+  // Only call for ads once the script has loaded
+  loadScript(global, 'https://www.google.com/adsense/search/ads.js', () => {
+
+    // Make the call for CSA ads
+    // Call the right product based on arguments passed
+    if (data['afsPageOptions'] != null && data['afshPageOptions'] == null) {
+      // AFS only
+
+      // Create the callback without any backfill options
+      afsAdblockOptions['adLoadedCallback'] = generateCallback(null, null);
+
+      global._googCsa('ads', afsPageOptions, afsAdblockOptions);
+
+    } else if (data['afsPageOptions'] == null
+      && data['afshPageOptions'] != null) {
+      // AFSH only
+
+      // Create the callback without any backfill options
+      afshAdblockOptions['adLoadedCallback'] = generateCallback(null, null);
+
+      global._googCsa('plas', afshPageOptions, afshAdblockOptions);
+
+    } else if (data['afsPageOptions'] != null
+      && data['afshPageOptions'] != null) {
+      // AFSh backfilled with AFS
+
+      // Create a callback with the AFS options
+      afshAdblockOptions['adLoadedCallback'] =
+          generateCallback(afsPageOptions, afsAdblockOptions);
+
+      global._googCsa('plas', afshPageOptions, afshAdblockOptions);
+    }
+  });
+
+}
+
+/**
+ * Create a callback function to resize the iframe and/or request backfill
+ * @param {*} backfillPageOptions AFS page options (if necessary)
+ * @param {*} backfillAdblockOptions AFS ad unit options (if necessary)
+ * @return {function(string,boolean)} The callback function
+ * @visibleForTesting
+ */
+export function generateCallback(backfillPageOptions, backfillAdblockOptions) {
+  /**
+   * CSA callback function to resize the iframe and/or request backfill
+   * @param {string} containerName
+   * @param {boolean} adsLoaded
+   */
+  const resizeIframe = function(containerName, adsLoaded) {
```

Comment:
instead of return a function, we typically do:

```
afshAdblockOptions['adLoadedCallback'] = resizeIframe.bind(null, afsPageOptions, afsAdblockOptions);

function resizeIframe(backfillPageOptions, backfillAdblockOptions, containerName, adsLoaded) {
  ...
}
```